chatterbot/adapters/logic/multi_adapter.py

```diff
@@ -24,9 +48,11 @@ def process(self, statement):
         result = None
         max_confidence = -1
 
-        for adapter in self.adapters:
-            if adapter.can_process(statement):
-                confidence, output = adapter.process(statement)
+        usable_adapters = self.get_usable_adapters(statement)
+
+        # with ProcessPoolExecutor(max_executors) as executor:
+        with ThreadPoolExecutor(max_workers=len(usable_adapters)) as executor:
+            for confidence, output, adapter in executor.map(self.execute_adapter_process, usable_adapters):
```

Comment:
I believe since this is happening sequentially (locked by the for loop), this isn't actually giving you any extra performance. Unless `.map` doesn't guarantee ordering, in which case this makes sense.